File: top-interview-150/c++/125_valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        // remove all non-alnum
        s.erase(std::remove_if(s.begin(), s.end(), 
        []( auto const& c ) -> bool { return !std::isalnum(c); } ), s.end());
        // to_lower
        std::transform(s.begin(), s.end(), s.begin(),
        [](unsigned char c){ return std::tolower(c); });
        //cout << s.size() << endl;
        //cout << s << endl;
        int length = s.size();
        for(int i=0; i<(length/2); i++)
        {
            if (s[i] != s[length-i-1]) return false;
        }
        return true;
    }
};

File: top-interview-150/python/125_valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        #print(s)
        import re, string
        s = re.sub('[\W_]+', '', s)
        s = s.lower()
        #print(s)
        length = len(s)
        bar = int(length / 2)
        #print(bar)
        for i in range(0, bar):
            #print(i)
            if s[i] != s[length-i-1]:
                return False
        return True
        